pkg/simple_cache/lib/src/providers/redis.dart

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' show SocketException;
+import 'package:dartis/dartis.dart';
+import 'package:retry/retry.dart';
+import '../../cache_provider.dart';
+
+class _RedisContext {
+  final Connection connection;
+  final Client client;
+  final Commands<String, List<int>> commands;
+  _RedisContext({this.connection, this.client, this.commands});
+}
+
+typedef _Op<T> = Future<T> Function(Commands<String, List<int>>);
+
+class RedisCacheProvider extends CacheProvider<List<int>> {
+  final String _connectionString;
+  final RetryOptions _retryOptions;
+  final Duration _connectTimeLimit;
+  final Duration _commandTimeLimit;
+
+  Future<_RedisContext> _context;
+  bool _isClosed = false;
+
+  RedisCacheProvider(
+    this._connectionString, {
+    RetryOptions retryOptions = const RetryOptions(),
```

Comment:
I'm a bit worried about the default settings. Normally we want to have a cache that answers the requests fast, and if I understand it correctly, these settings will (a) retry it many times, and (b) timeout after 5 seconds, which is way above the usual time for our non-cached request serving. I'd probably start with a 1-retry, <200ms setting first.

Reply:
Good point, this is just caching, we should be relaxed about retrying.. and fail fast instead.